src/cell2sentence/csmodel.py

```diff
@@ -164,17 +170,26 @@ def data_collator(examples):
         print(f"Starting training. Output directory: {output_dir}")
 
         # Perform dataset split
-        split_ds_dict, data_split_indices_dict = train_test_split_arrow_ds(formatted_hf_ds)
+        if data_split_indices_dict is None:
+            split_ds_dict, data_split_indices_dict = train_test_split_arrow_ds(formatted_hf_ds)
+        else:
+            # Dataset split indices supplied by user, split formatted arrow dataset accordingly
+            train_ds = formatted_hf_ds.select(data_split_indices_dict["train"])
+            val_ds = formatted_hf_ds.select(data_split_indices_dict["val"])
+            ds_dict_object = {
+                "train": train_ds,
+                "validation": val_ds,
+            }
+            split_ds_dict = DatasetDict(ds_dict_object)
         with open(os.path.join(output_dir, 'data_split_indices_dict.pkl'), 'wb') as f:
             pickle.dump(data_split_indices_dict, f)
         
         train_dataset = split_ds_dict["train"]
         eval_dataset = split_ds_dict["validation"]
-        if max_eval_samples is not None:
+        if (max_eval_samples is not None) and (max_eval_samples < eval_dataset.num_rows):
```

Comment:
Instead of not using `max_eval_samples` at all if it's greater than `eval_dataset.num_rows`, it's probably better to just use the min of the two:
`max_eval_samples = min(max_eval_samples, eval_dataset.num_rows)`

Reply:
In the case that `max_eval_samples` is greater than `eval_dataset.num_rows`, I think that using the min of the two would be equivalent to leaving the eval_dataset as is. The random sample would sample the entire list of indices, so the eval_dataset would stay the same.

If we want to keep the behavior of saving the sample indices to disk then maybe we could use min and still have the if conditional run, but otherwise I believe things would be equivalent to the if conditional not running. Let me know what you think!

Reply:
Ah fair enough, that makes sense. Looks good then.